userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
```

Comment:
Moved this to `rclcpp` and `rclcpp::executors`:

https://github.com/ros2/rclcpp/pull/20/files#diff-690567624ee368716009118718985564R81
https://github.com/ros2/rclcpp/pull/20/files#diff-294cbb523fb3d56a1df2a0a7ff0beb9bR31